metricbeat/helper/prometheus/prometheus.go

```diff
@@ -142,13 +153,19 @@ func (p *prometheus) GetProcessedMetrics(mapping *MetricsMapping) ([]common.MapS
 			// Convert labels
 			labels := common.MapStr{}
 			keyLabels := common.MapStr{}
+
 			for k, v := range allLabels {
 				if l, ok := mapping.Labels[k]; ok {
 					if l.IsKey() {
 						keyLabels.Put(l.GetField(), v)
 					} else {
 						labels.Put(l.GetField(), v)
 					}
+				} else if storeAllLabels {
+					// if label for this metric is not found at the label mappings but
+					// it is configured to store any labels found, make it so
+					// TODO dedot
+					labels.Put(labelsLocation+"."+k, v)
```

Comment:
I think prometheus labels don't contain dots?

Reply:
that's interesting, I will need to test but prometheus is copying labels + values from kubernetes, which might contain dots.

I'm not sure if it will code dots into something else. I'll keep the TODO until I test